redbox-core/redbox/loader/loaders.py

```diff
@@ -72,7 +72,10 @@ def extract_metadata(self) -> dict:
         original_metadata = chunks[0]["metadata"] if chunks else {}
         first_thousand_words = "".join(chunk["text"] for chunk in chunks)[:10_000]
 
-        metadata = self.create_file_metadata(first_thousand_words, original_metadata=original_metadata)
+        try:
+            metadata = self.create_file_metadata(first_thousand_words, original_metadata=original_metadata)
+        except TypeError:
+            metadata = GeneratedMetadata(name=original_metadata.get("filename"))
```

Comment:
Does original_metadata always have filename?

Reply:
its not guaranteed but it is the most likely https://docs.unstructured.io/api-reference/api-services/document-elements#common-metadata-fields